core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooReaderWriter.java

```diff
@@ -138,6 +128,24 @@ public void recursiveCopyPersistentOverwrite(String source, String destination)
     }, e -> e.code() == Code.BADVERSION); // always retry if bad version
   }
 
+  public byte[] createPublicOrMutate(String zPath, byte[] createValue, Mutator mutator)
```

Comment:
Feel free to ignore this comment, its very subjective. It would be nice if both mutate methods had the same prefix, for the case when completing in IDE.  Also, if there was javadoc I would drop public from the method name and put that in javadoc.  With those two thoughts, I like the name `mutateOrCreate`.

Reply:
Alike minds... I had the same thoughts and shortened the name after adding the javadoc. I'm still a bit torn on `mutateOrCreate` or `createOrMutate`, though... I'm not sure I like the implication that we might try to mutate first, if I call it `mutateOrCreate`, but the argument from auto-complete is a compelling one, so I'll just go with that. :smiley_cat:

Reply:
I was thinking about the ordering too.  I think it does not matter, because the implication of the OR is that it will only do one of create or mutate.  How it only does one is an implementation detail.  So the fact that the implementation does create first is not important to the method name.